src/safir/dependencies/logger.py

```diff
@@ -0,0 +1,57 @@
+"""Logger dependency for FastAPI.
+
+Provides a :py:mod:`structlog` logger as a FastAPI dependency.  The logger
+will incorporate information from the request in its bound context.
+"""
```

Comment:
Does this need ``__future__`` ``annotations``?

Reply:
You can't use `from __future__ import annotations` with FastAPI handlers and dependencies currently because FastAPI and pydantic don't fully support deferred annotations. This is a whole thing, which is being discussed in Python upstream and there will probably be changes. In the meantime, adopting this feature has been deferred until no earlier than 3.11.